hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMBatchKeyDeleteRequest.java

```diff
@@ -0,0 +1,226 @@
+import org.apache.hadoop.ozone.security.acl.IAccessAuthorizer;
+import org.apache.hadoop.ozone.security.acl.OzoneObj;
+import org.apache.hadoop.util.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.*;
+
+import static org.apache.hadoop.ozone.om.exceptions.OMException.ResultCodes.KEY_NOT_FOUND;
+import static org.apache.hadoop.ozone.om.lock.OzoneManagerLock.Resource.BUCKET_LOCK;
+
+/**
+ * Handles DeleteKey request.
+ */
+public class OMBatchKeyDeleteRequest extends OMKeyRequest {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(OMBatchKeyDeleteRequest.class);
+
+  public OMBatchKeyDeleteRequest(OMRequest omRequest) {
+    super(omRequest);
+  }
+
+  @Override
+  public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
+    DeleteBatchKeyRequest deleteKeyRequest =
+        getOmRequest().getDeleteBatchKeyRequest();
+    Preconditions.checkNotNull(deleteKeyRequest);
+    List<KeyArgs> newKeyArgsList = new ArrayList<>();
+    for (KeyArgs keyArgs : deleteKeyRequest.getKeyArgsList()) {
+      newKeyArgsList.add(
+          keyArgs.toBuilder().setModificationTime(Time.now()).build());
+    }
+    DeleteBatchKeyRequest newDeleteKeyRequest = DeleteBatchKeyRequest
+        .newBuilder().addAllKeyArgs(newKeyArgsList).build();
+
+    return getOmRequest().toBuilder()
+        .setDeleteBatchKeyRequest(newDeleteKeyRequest)
+        .setUserInfo(getUserInfo()).build();
+  }
+
+  @Override
+  @SuppressWarnings("methodlength")
+  public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
+      long trxnLogIndex, OzoneManagerDoubleBufferHelper omDoubleBufferHelper) {
+    DeleteBatchKeyRequest deleteKeyRequest =
+        getOmRequest().getDeleteBatchKeyRequest();
+
+    List<KeyArgs> deleteKeyArgsList = deleteKeyRequest.getKeyArgsList();
+    IOException exception = null;
+    OMClientResponse omClientResponse = null;
+    Result result = null;
+
+    OMMetrics omMetrics = ozoneManager.getMetrics();
+    omMetrics.incNumKeyDeletes();
+    List<OmKeyInfo> omKeyInfoList = new ArrayList<>();
+    Set<String> acquiredLockSet = new HashSet<>();
+    Map<String, String> auditMap = null;
+    String volumeName = "";
+    String bucketName = "";
+    String keyName = "";
+
+    AuditLogger auditLogger = ozoneManager.getAuditLogger();
+    OzoneManagerProtocolProtos.UserInfo userInfo =
+        getOmRequest().getUserInfo();
+
+    OMResponse.Builder omResponse = OmResponseUtil.getOMResponseBuilder(
+        getOmRequest());
+    OMMetadataManager omMetadataManager = ozoneManager.getMetadataManager();
+    try {
+      for (KeyArgs deleteKeyArgs : deleteKeyArgsList) {
+        acquiredLockSet.add(deleteKeyArgs.getVolumeName() + "_" +
+            deleteKeyArgs.getBucketName());
+      }
+
+      for (String volumeAndVolume : acquiredLockSet) {
+        omMetadataManager.getLock().acquireWriteLock(
```

Comment:
Note when the batch size is large, this could hold the lock for a longer period of time compare with the individual lock and release.

Reply:
I had fixed this part by getting the lock separately when the key is deleting.